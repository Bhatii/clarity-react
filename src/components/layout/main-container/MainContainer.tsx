import * as React from "react";
import {ResponsiveNavCodes} from "../nav/ResponsiveNavCodes";
import Header from "../nav/Header";

export type MainContainerProps = {
    primaryNav?: React.ReactNode;
    secondaryNav?: React.ReactNode;
};

type mainContainerState = {
    navList: number[];
    leftNavOpen: boolean;
    rightNavOpen: boolean;
};

export default class MainContainer extends React.PureComponent<
    MainContainerProps
> {
    state: mainContainerState;

    constructor(props: any) {
        super(props);
        let navList: number[] = [];
        if (props.primaryNav) {
            navList.push(ResponsiveNavCodes.NAV_LEVEL_1);
        }
        if (props.secondaryNav) {
            navList.push(ResponsiveNavCodes.NAV_LEVEL_2);
        }
        this.state = {
            navList: navList,
            leftNavOpen: false,
            rightNavOpen: false,
        };
    }

    handleHamburgerToggle() {
        const {leftNavOpen} = this.state;
        this.setState({leftNavOpen: !leftNavOpen});
    }

    handleRightSideToggle() {
        const {rightNavOpen} = this.state;
        this.setState({rightNavOpen: !rightNavOpen});
    }

    closeAll() {
        this.setState({leftNavOpen: false, rightNavOpen: false});
    }

    getClassList() {
        let classList: string[] = [ResponsiveNavCodes.MAIN_CONTAINER_CLASS];
        const {leftNavOpen, rightNavOpen} = this.state;
        if (leftNavOpen) {
            classList.push(ResponsiveNavCodes.NAV_CLASS_HAMBURGER_MENU);
        }
        if (rightNavOpen) {
            classList.push(ResponsiveNavCodes.NAV_CLASS_OVERFLOW_MENU);
        }
        return classList;
    }

    render() {
        const {navList} = this.state;
        const {primaryNav, secondaryNav} = this.props;
        return (
            <div className={this.getClassList().join(" ")}>
                <Header
                    navList={navList}
                    onHamburgerToggle={this.handleHamburgerToggle.bind(this)}
                    onRightSideToggle={this.handleRightSideToggle.bind(this)}
                    onCloseAll={this.closeAll.bind(this)}
                >
                    <div className="branding">
                        <a href="#" className="nav-link">
                            <span className="logo dell-emc-logo" />
                            <span className="title">ECS Flex</span>
                        </a>
                    </div>
                    {primaryNav}
                    <div className="header-actions" />
                </Header>
                {secondaryNav}
                <div className="content-container">{this.props.children}</div>
            </div>
        );
    }
}
