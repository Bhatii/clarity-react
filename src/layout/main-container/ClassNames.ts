export class ClassNames {
    public static CONTAINER_CLASS: string = "main-container";
}
