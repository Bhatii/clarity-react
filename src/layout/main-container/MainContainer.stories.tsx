import * as React from 'react';
import {storiesOf} from '@storybook/react';
import {MainContainer} from ".";
import {Nav, NavLevel, NavLink, NavType} from "../nav";

storiesOf('MainContainer', module)
    .add('Main Container', () =>
        <MainContainer
            title="Storybook"
            headerNav={
                <Nav navLevel={NavLevel.PRIMARY} navType={NavType.HEADER}>
                    <NavLink>Link 1</NavLink>
                    <NavLink>Link 2</NavLink>
                </Nav>
            }
            subNav={
                <Nav navLevel={NavLevel.SECONDARY} navType={NavType.SUB}>
                    <NavLink>Link 3</NavLink>
                    <NavLink>Link 4</NavLink>
                </Nav>
            }
        />
    )
;
