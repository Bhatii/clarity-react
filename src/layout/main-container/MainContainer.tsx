import * as React from 'react';
import * as utils from "../../utils";
import {Header, ClassNames as NavClassNames} from "../nav";
import {ClassNames} from ".";

export type MainContainerProps = {
    title: string;
    primaryNav?: React.ReactNode;
    secondaryNav?: React.ReactNode;
};

const initialState = {
    leftNavOpen: false,
    rightNavOpen: false,
};

type MainContainerState = Readonly<typeof initialState>;

export class MainContainer extends React.PureComponent<MainContainerProps> {
    readonly state: MainContainerState = initialState;

    private static propIsSet(prop: any): boolean {
        return (typeof prop !== "undefined" && prop !== null);
    }

    handleHamburgerToggle = () => {
        const {leftNavOpen} = this.state;
        this.setState({leftNavOpen: !leftNavOpen});
    };

    handleRightSideToggle = () => {
        const {rightNavOpen} = this.state;
        this.setState({rightNavOpen: !rightNavOpen});
    };

    closeAll = () => {
        this.setState({leftNavOpen: false, rightNavOpen: false});
    };

    getClassList(): (string | undefined)[] {
        const {leftNavOpen, rightNavOpen} = this.state;
        return [
            ClassNames.CONTAINER_CLASS,
            (leftNavOpen ? NavClassNames.HAMBURGER_MENU : undefined),
            (rightNavOpen ? NavClassNames.OVERFLOW_MENU : undefined)
        ];
    }

    render() {
        const {children, primaryNav, secondaryNav, title} = this.props;
        return (
            <div className={utils.classNames(this.getClassList())}>
                <Header
                    isNavLevel1OnPage={MainContainer.propIsSet(primaryNav)}
                    isNavLevel2OnPage={MainContainer.propIsSet(secondaryNav)}
                    onHamburgerToggle={this.handleHamburgerToggle}
                    onRightSideToggle={this.handleRightSideToggle}
                    onCloseAll={this.closeAll}
                >
                    <div className="branding">
                        <a href="#" className="nav-link">
                            <span className="logo dell-emc-logo"/>
                            <span className="title">
                                {title}
                            </span>
                        </a>
                    </div>
                    {primaryNav}
                    <div className="header-actions"/>
                </Header>
                {secondaryNav}
                <div className="content-container">
                    {children}
                </div>
            </div>
        );
    }
}
