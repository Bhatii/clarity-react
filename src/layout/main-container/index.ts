export * from "./MainContainer";
export * from "./ClassNames";
