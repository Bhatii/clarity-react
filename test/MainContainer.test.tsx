import {MainContainer} from "../src/layout/main-container/MainContainer";
import {shallow} from "enzyme";
import * as React from "react";

describe("<MainContainer /> rendering", () => {
    it("renders correctly", () => {
        const wrapper = shallow(<MainContainer title="test" />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render one <Header />", () => {
        const wrapper = shallow(<MainContainer title="test" />);
        expect(wrapper.find("Header")).toHaveLength(1);
    });
});
