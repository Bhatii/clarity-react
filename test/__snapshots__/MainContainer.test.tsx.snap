// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContainer /> rendering renders correctly 1`] = `
<div
  className="main-container"
>
  <Header
    onCloseAll={[Function]}
    onHamburgerToggle={[Function]}
    onRightSideToggle={[Function]}
    primaryShown={false}
    secondaryShown={false}
  >
    <div
      className="branding"
    >
      <a
        className="nav-link"
        href="#"
      >
        <span
          className="logo dell-emc-logo"
        />
        <span
          className="title"
        >
          test
        </span>
      </a>
    </div>
    <div
      className="header-actions"
    />
  </Header>
  <div
    className="content-container"
  />
</div>
`;
